src/App.tsx

```diff
@@ -1,61 +1,117 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User, Todo } from './components/types';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(
+    todosFromServer.map(todo => ({
+      ...todo,
+      user: getUserById(todo.userId) as User,
+    })),
+  );
+
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserIdError] = useState(false);
+  const [userId, setUserId] = useState(0);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setTitleError(!title);
+    setUserIdError(!userId);
+
+    if (!title || !userId) {
```

Comment:
```suggestion
    if (!title.trim() || !userId) {
```
<img width="213" alt="image" src="https://github.com/user-attachments/assets/3fe416f3-c682-4e2f-9f92-fcc742db8ee8">
to fix empty todos creating